test/jdk/java/awt/Graphics2D/DrawString/IgnoredWhitespaceTest.java

```diff
@@ -0,0 +1,185 @@
+        test(image, g2d, font);
+
+        g2d.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON);
+        test(image, g2d, font);
+
+        g2d.setRenderingHint(RenderingHints.KEY_FRACTIONALMETRICS, RenderingHints.VALUE_FRACTIONALMETRICS_ON);
+        test(image, g2d, font);
+
+        Font kerningFont = font.deriveFont(Map.of(TextAttribute.KERNING, TextAttribute.KERNING_ON));
+        test(image, g2d, kerningFont);
+
+        Font physicalFont = getPhysicalFont(40);
+        if (physicalFont != null) {
+            test(image, g2d, physicalFont);
+        }
+
+        g2d.dispose();
+    }
+
+    private static void test(BufferedImage image, Graphics2D g2d, Font font) {
+        test(image, g2d, font, "XXXXX", "\t\t\t\t\tXXXXX");
+        test(image, g2d, font, "XXXXX", "\tX\tX\tX\tX\tX\t");
+        test(image, g2d, font, "XXXXX", "\r\r\r\r\rXXXXX");
+        test(image, g2d, font, "XXXXX", "\rX\rX\rX\rX\rX\r");
+        test(image, g2d, font, "XXXXX", "\n\n\n\n\nXXXXX");
+        test(image, g2d, font, "XXXXX", "\nX\nX\nX\nX\nX\n");
+    }
+
+    private static void test(BufferedImage image, Graphics2D g2d, Font font, String reference, String text) {
+
+        g2d.setFont(font);
+        FontRenderContext frc = g2d.getFontRenderContext();
+        int w = image.getWidth();
+        int h = image.getHeight();
+        int x = w / 2;
+        int y = h / 2;
+
+        g2d.setColor(Color.WHITE);
+        g2d.fillRect(0, 0, w, h);
+        g2d.setColor(Color.BLACK);
+        g2d.drawString(reference, x, y);
+        Rectangle expected = findTextBoundingBox(image);
+
+        g2d.setColor(Color.WHITE);
+        g2d.fillRect(0, 0, w, h);
+        g2d.setColor(Color.BLACK);
+        g2d.drawString(text, x, y);
+        Rectangle actual = findTextBoundingBox(image);
+        assertEqual(expected, actual, text);
+
+        g2d.setColor(Color.WHITE);
+        g2d.fillRect(0, 0, w, h);
+        g2d.setColor(Color.BLACK);
+        g2d.drawString(new AttributedString(text, Map.of(TextAttribute.FONT, font)).getIterator(), x, y);
+        actual = findTextBoundingBox(image);
+        assertEqual(expected, actual, text);
+
+        g2d.setColor(Color.WHITE);
+        g2d.fillRect(0, 0, w, h);
+        g2d.setColor(Color.BLACK);
+        g2d.drawChars(text.toCharArray(), 0, text.length(), x, y);
+        actual = findTextBoundingBox(image);
+        assertEqual(expected, actual, text);
+
+        g2d.setColor(Color.WHITE);
+        g2d.fillRect(0, 0, w, h);
+        g2d.setColor(Color.BLACK);
+        g2d.drawGlyphVector(font.createGlyphVector(frc, text), x, y);
+        actual = findTextBoundingBox(image);
+        assertEqual(expected, actual, text);
+    }
+
+    private static void assertEqual(Rectangle r1, Rectangle r2, String text) {
+        if (!r1.equals(r2)) {
+            String escaped = text.replace("\r", "\\r").replace("\n", "\\n").replace("\t", "\\t");
```

Comment:
I would format this way, so that each `replace` stands out in the chain of calls.
```suggestion
            String escaped = text.replace("\r", "\\r")
                                 .replace("\n", "\\n")
                                 .replace("\t", "\\t");
```